src/forms/renderers/ConnectorSelect/AutoComplete.tsx

```diff
@@ -91,45 +97,25 @@ export const ConnectorAutoComplete = (
             getOptionLabel={getOptionLabel ?? ((option) => option.label)}
             sx={{
                 marginTop: 2,
-                borderColor: errors.length > 0 ? 'red' : null,
             }}
-            renderInput={({ inputProps, InputProps }: any) => {
+            renderInput={({ inputProps, InputProps }) => {
                 return (
-                    <Box
-                        sx={{
-                            '.MuiBox-root + .MuiInput-root > input': {
-                                textIndent: '20px',
-                            },
-                        }}
-                    >
-                        {inputProps.value !== '' ? (
-                            <Box
-                                sx={{
-                                    position: 'absolute',
-                                    left: 0,
-                                    top: 22,
-                                }}
-                            >
-                                <ConnectorName path="" connector="" />
-                            </Box>
-                        ) : null}
-
-                        <Input
-                            style={{ width: '100%' }}
-                            type="text"
-                            inputProps={inputProps}
-                            inputRef={InputProps.ref}
-                            autoFocus={appliedUiSchemaOptions.focus}
-                            disabled={!enabled}
-                        />
-                    </Box>
+                    <ConnectorInput
+                        inputProps={inputProps}
+                        InputProps={InputProps}
+                        appliedUiSchemaOptions={appliedUiSchemaOptions}
+                        enabled={enabled}
+                        currentOption={currentOption}
+                    />
                 );
             }}
-            renderOption={(renderOptionProps: any, option) => {
+            renderOption={(renderOptionProps, option) => {
                 return (
-                    <Box component="li" {...renderOptionProps}>
-                        <ConnectorName path="" connector={option.label} />
-                    </Box>
+                    <ConnectorOption
+                        renderOptionProps={renderOptionProps}
+                        option={option}
+                        key={option.label}
+                    />
                 );
```

Comment:
Removing the input and options into components to reduce the "size" of the renderer.